snooty/parser.py

```diff
@@ -644,6 +650,99 @@ def handle_tabset(self, node: n.Directive) -> None:
+        expected_child_opt_name = "wayfinding-option"
+        expected_child_desc_name = "wayfinding-description"
+        expected_children_names = {expected_child_opt_name, expected_child_desc_name}
+        wayfinding_name = "wayfinding"
+
+        valid_opts: List[n.Node] = []
+        valid_desc: n.Node | None = None
+        used_ids: Set[str] = set()
+
+        # Validate children
+        for child in node.children:
+            child_line_start = child.start[0]
+
+            invalid_child = None
+            if not isinstance(child, n.Directive):
+                # Catches additional unwanted types like Paragraph
+                invalid_child = child.type
+            elif not child.name in expected_children_names:
+                invalid_child = child.name
+
+            if invalid_child:
+                self.diagnostics.append(
+                    InvalidChild(
+                        invalid_child,
+                        wayfinding_name,
+                        f"{expected_child_opt_name} or {expected_child_desc_name}",
+                        child_line_start,
+                    )
+                )
+                continue
+
+            # Type is ambiguous now despite if statements above
+            assert isinstance(child, n.Directive)
+            if child.name == expected_child_desc_name:
+                valid_desc = child
+                continue
+
+            option_id = child.options.get("id")
+
+            if not (child.argument and option_id):
+                # Don't append diagnostic since docutils should already
+                # complain about missing argument and ID option
+                continue
+
+            if not option_id in expected_options_dict:
+                self.diagnostics.append(
+                    UnknownWayfindingOption(option_id, child_line_start)
+                )
+                continue
+
+            if option_id in used_ids:
+                self.diagnostics.append(
+                    DuplicateWayfindingOption(option_id, child_line_start)
+                )
+                continue
+
+            option_details = expected_options_dict[option_id]
+            child.options["title"] = option_details.title
+            child.options["language"] = option_details.language
+            valid_opts.append(child)
+            used_ids.add(option_id)
+
+        def sort_key(node: n.Node) -> tuple[bool, str, str]:
+            # All children should be directives; keeping Node type for type compatibility of new children
+            assert isinstance(node, n.Directive)
+            # Associate the child node with the actual wayfinding option
+            wayfinding_option = expected_options_dict[node.options["id"]]
+            return (
+                not wayfinding_option.show_first,
+                wayfinding_option.language,
+                wayfinding_option.title,
+            )
+
+        new_children = sorted(valid_opts, key=sort_key)
```

Comment:
minor. `new_children` is typed as `any` here. can we do a `valid_opts.sort()` ?

Reply:
`valid_opts.sort()` should be safe -- it's a behavioral change, but the only subsequent use of `valid_opts` is checking its size, so it's all good. Plus, in-place sort

Reply:
Whoops, thanks for catching. Made the change